packages/web-components/src/components/ic-badge/ic-badge.tsx

```diff
@@ -0,0 +1,225 @@
+    this.visible = false;
+  }
+
+  private setBadgeColour = () => {
+    let colorRGBA = null;
+    if (this.customColor !== null) {
+      const firstChar = this.customColor.slice(0, 1);
+      if (firstChar === "#") {
+        colorRGBA = hexToRgba(this.customColor);
+      } else if (firstChar.toLowerCase() === "r") {
+        colorRGBA = rgbaStrToObj(this.customColor);
+      }
+      this.customColorRGBA = colorRGBA;
+      this.el.style.backgroundColor = `rgba(${colorRGBA.r.toString()}, ${colorRGBA.g.toString()}, ${colorRGBA.b.toString()}, ${colorRGBA.a.toString()})`;
+    }
+  };
+
+  private getBadgeForeground = () => {
+    let red: number;
+    let green: number;
+    let blue: number;
+
+    switch (this.variant) {
+      case "custom":
+        red = this.customColorRGBA.r;
+        green = this.customColorRGBA.g;
+        blue = this.customColorRGBA.b;
+        break;
+      case "error":
+        red = parseInt(getCssProperty("--ic-status-error-r"));
+        green = parseInt(getCssProperty("--ic-status-error-g"));
+        blue = parseInt(getCssProperty("--ic-status-error-b"));
+        break;
+      case "success":
+        red = parseInt(getCssProperty("--ic-status-success-r"));
+        green = parseInt(getCssProperty("--ic-status-success-g"));
+        blue = parseInt(getCssProperty("--ic-status-success-b"));
+        break;
+      case "warning":
+        red = parseInt(getCssProperty("--ic-status-warning-r"));
+        green = parseInt(getCssProperty("--ic-status-warning-g"));
+        blue = parseInt(getCssProperty("--ic-status-warning-b"));
+        break;
+      case "info": {
+        const info = hexToRgba(getCssProperty("--ic-status-info"));
+        red = info.r;
+        green = info.g;
+        blue = info.b;
+        break;
+      }
+      case "neutral": {
+        const neutral = hexToRgba(getCssProperty("--ic-architectural-500"));
+        red = neutral.r;
+        green = neutral.g;
+        blue = neutral.b;
+        break;
+      }
+      case "light": {
+        const light = hexToRgba(getCssProperty("--ic-architectural-40"));
+        red = light.r;
+        green = light.g;
+        blue = light.b;
+        break;
+      }
+    }
+
+    const brightness = (red * 299 + green * 587 + blue * 114) / 1000;
+    this.foregroundColour =
+      brightness > 133.3505
+        ? IcThemeForegroundEnum.Dark
+        : IcThemeForegroundEnum.Light;
+  };
+
+  private getTextLabel = () => {
+    let label;
+    if (this.textLabel !== null) {
+      if (this.maxNumber !== null) {
+        label =
+          this.textLabel > this.maxNumber
+            ? this.maxNumber + "+"
+            : this.textLabel;
+      } else {
+        label = this.textLabel;
+      }
+    }
+    return label;
+  };
+
+  private setAccessibleLabel = () => {
+    if (this.accessibleLabel !== null && getParentElement(this.el) !== null) {
+      getParentElement(this.el).ariaLabel =
+        "badge displaying " + this.accessibleLabel;
```

Comment:
On tabs, this aria label replaces the label of the tab. E.g. in the "Position inline" story, it just says "badge displaying undefined" and doesn't say the word "Small" which is the visible label of the tab